src/io/node.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-console */
+import process from 'node:process'
+import chalk from 'chalk'
+import nodeReleases from 'node-releases/data/release-schedule/release-schedule.json'
+
+function getLatestNodeVersion() {
+  const versions = Object.keys(nodeReleases)
+  const latestVersion = versions[versions.length - 1]
+  const versionWithoutV = latestVersion.slice(1)
+  const nodeVersion = versionWithoutV.split('.')[0]
+  return nodeVersion
+}
+
+function getNodeVersion() {
+  const nodeVersionWithoutV = process.version.slice(1)
+  const nodeVersion = nodeVersionWithoutV.split('.')[0]
+  return nodeVersion
+}
+
+function checkNode() {
+  const nodeVersion = getNodeVersion()
+  const nodeVersionData = nodeReleases[`v${nodeVersion}` as keyof typeof nodeReleases]
+  let result = false
+  if (nodeVersionData) {
+    const endDate = new Date(nodeVersionData.end)
+    const currentDate = new Date()
+    const isNodeVersionSupported = currentDate < endDate
+    if (isNodeVersionSupported) {
+      console.log(chalk.green(`Your node version (${nodeVersion}) is supported until ${nodeVersionData.end}.`))
+      result = true
+    }
+    else {
+      console.log(chalk.yellow(`Your node version (${nodeVersion}) is no longer supported since ${nodeVersionData.end}.`))
+      result = false
+    }
+  }
+  else if (nodeVersion > getLatestNodeVersion()) {
```

Comment:
When might this happen?

Reply:
I can see two unlikely cases:

1. Someone has ndc installed, updates their node environment after 2 or 3 years, but does not make an `npm update` or a new `npm install` in the project where they use ndc.
2. Or the newly introduced package ‘node-releases’ is no longer updated. The EOL information is integrated directly into that package and is not queried via an API.

This is not ideal, but it works for now and both variants are not that likely.

Reply:
Okay, i know. If `node-releases` is no longer updated, we can fork it to maintain.
So the log message can be changed to prompt users to update `ndc`?

Reply:
It can be changed to `semver.gt`.

Reply:
Nice. I'll take care of it (and the other comments), but may not find the time for it for a few days.

Reply:
> It can be changed to semver.gt.

Done.

> So the log message can be changed to prompt users to update ndc?

Yes, and done.